Modules/readline.c

```diff
@@ -1026,7 +1077,18 @@ on_startup_hook(void)
 {
     int r;
     PyGILState_STATE gilstate = PyGILState_Ensure();
-    r = on_hook(readlinestate_global->startup_hook);
+    PyObject* mod = PyState_FindModule(&readlinemodule);
+    if (mod == NULL) {
+        if (!PyErr_Occurred()) {
+            PyErr_SetString(PyExc_ImportError, "readline import initialization failure during startup hook");
+        }
+        readline_cleanup();
+        PyGILState_Release(gilstate);
+        return -1;
+    }
+    Py_INCREF(mod);
+    r = on_hook(get_readline_state(mod)->startup_hook);
+    Py_DECREF(mod);
```

Comment:
```suggestion
    readlinestate *state = get_global_readline_state_with_error();
    if (state == NULL) {
        PyErr_SetString(PyExc_ImportError, "readline import initialization failure during startup hook");
        readline_cleanup();
        PyGILState_Release(gilstate);
        return -1;
    }
    r = on_hook(state->startup_hook);
```

You could also chain the exception, but I don't think that would add much value.

Reply:
Considered the point I make elsewhere about `get_hook_module_state()` and about calling `PyState_FindModule()` early in `setup_readline()`, we can go even simpler:

```suggestion
    readlinestate *state = get_hook_module_state();
    if (state == NULL) {
        PyGILState_Release(gilstate);
        return -1;
    }
    r = on_hook(state->startup_hook);
```

Note we don't set an exception at all.  That's because the exception won't get caught anywhere.  See how we call `PyErr_Clear()` in the error case of `on_completion_display_matches_hook()`.